### PR TITLE
#7 Task. MironovNik

### PR DESCRIPTION
**Pull request by MironovNick**
https://travis-ci.com/github/MironovNick/js-assignments

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -33,7 +33,17 @@
  *
  */
 function* get99BottlesOfBeer() {
-    throw new Error('Not implemented');
+    //throw new Error('Not implemented');
```
Comment (KostyaKov) [inline task/07-yield-tasks.js]: unnecessary comments need to be removed

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -33,7 +33,17 @@
...
+	for(let i = 99; i >=3; i--) {
```
Comment (KostyaKov) [inline task/07-yield-tasks.js]: it is necessary to observe indents between operators and after "for"

File: task/07-yield-tasks.js [hunk 3]
```diff
@@ -33,7 +33,17 @@
...
+		yield i + " bottles of beer on the wall, " + i + " bottles of beer.";
+		yield "Take one down and pass it around, " + (i-1) + " bottles of beer on the wall.";
+	}
+	yield '2 bottles of beer on the wall, 2 bottles of beer.';
+    yield 'Take one down and pass it around, 1 bottle of beer on the wall.';
```
Comment (KostyaKov) [inline task/07-yield-tasks.js:41]: it is necessary to observe indents

File: task/07-yield-tasks.js [hunk 4]
```diff
@@ -126,7 +166,30 @@ function* breadthTraversalTree(root) {
  *   [ 1, 3, 5, ... ], [ -1 ] => [ -1, 1, 3, 5, ...]
  */
 function* mergeSortedSequences(source1, source2) {
-    throw new Error('Not implemented');
+    //throw new Error('Not implemented');
+    let gen1 = source1();
+    let gen2 = source2();
+    let ob1 = gen1.next();
+    let ob2 = gen2.next();
+    while( !ob1.done || !ob2.done) {
```
Comment (KostyaKov) [inline task/07-yield-tasks.js]: it is necessary to observe indents

File: task/07-yield-tasks.js [hunk 5]
```diff
@@ -108,7 +136,19 @@ function* depthTraversalTree(root) {
  *
  */
 function* breadthTraversalTree(root) {
-    throw new Error('Not implemented');
+    //throw new Error('Not implemented');
+    let queue = [], queue2 = [];
```
Comment (KostyaKov) [inline task/07-yield-tasks.js:136]: it is necessary to observe indents

File: task/07-yield-tasks.js [hunk 6]
```diff
@@ -82,7 +101,16 @@ function* getFibonacciSequence() {
...
 function* depthTraversalTree(root) {
-    throw new Error('Not implemented');
+    //throw new Error('Not implemented');
+    let stack = [];
+	stack.push(root)
+	while(stack.length) {
+		let tmpNode = stack.pop();
+		yield tmpNode;
+		if(tmpNode.children)
+			for( let i = tmpNode.children.length-1; i >= 0; i--)
```
Comment (KostyaKov) [inline task/07-yield-tasks.js]: it is necessary to observe indents

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**